src/client/datascience/raw-kernel/rawKernel.ts

```diff
@@ -160,10 +160,28 @@ export class RawKernel implements Kernel.IKernelConnection {
     public reconnect(): Promise<void> {
         throw new Error('Reconnect is not supported.');
     }
-    public interrupt(): Promise<void> {
+    public async interrupt(): Promise<void> {
+        // Send a kernel interrupt request to the real process only for our python kernels.
+
         // Send this directly to our kernel process. Don't send it through the real kernel. The
         // real kernel will send a goofy API request to the websocket.
-        return this.kernelProcess.interrupt();
+        if (this.kernelProcess.canInterrupt) {
+            return this.kernelProcess.interrupt();
+        } else if (this.kernelProcess.kernelConnectionMetadata.kernelSpec.interrupt_mode === 'message') {
+            // traceInfo(`Interrupting kernel with a shell message`);
+            // const jupyterLab = require('@jupyterlab/services') as typeof import('@jupyterlab/services');
+            // const msg = (jupyterLab.KernelMessage.createMessage({
+            //     msgType: 'interrupt_request' as any,
+            //     channel: 'shell',
+            //     username: this.realKernel.username,
+            //     session: this.realKernel.clientId,
+            //     content: {}
+            // }) as any) as KernelMessage.IShellMessage<'inspect_request'>;
+            // await this.realKernel.sendShellMessage<'inspect_request'>(msg, true, true).done;
```

Comment:
This code seems to be right, however I can't find any kernel that implements this even though the protocol supports this.
Hence not sure whether we should enable this or not.

Worse case nothing happens
Commented this out so we can discuss this.
Personally feel its ok to enable this, if there are kernels that support message interrupts (for windows or even unix)

Reply:
Yeah seems okay to me.

Reply:
Ok to enable or  keep it commented out.

Reply:
Ok to enable. It doesn't hurt anything AFAICT

Reply:
Code looks like it's for 'inspect_request'? Isn't that for variables?

Reply:
Sorry,that part was for generics to work. Jupyter Lab NPM package doesn't support the interrupt_message request in their TS Types.
I changed it and used `any`.
The actual payload however is for interrupt request.